lightning/src/util/persist.rs

```diff
@@ -194,3 +206,857 @@ where
 	}
 	Ok(res)
 }
+enum KVStoreUpdatingPersisterError<'a> {
```

Comment:
Probably a good idea, I added it to make it easy to add some more error context, we could use that in both approaches. Not that you'd be likely to recover from one of these, but it'd hopefully help you see what's wrong. I'll work on it.

Reply:
I removed these. Couldn't solve the type puzzle in 1.48 / no-std without using literals, which made them totally pointless since they really just helped carry context. 

In the end, it's for the best. Adding error logging at all the points where we return an error should give a human all the context needed, and these errors are not really recoverable (well, anything is if you try hard enough, but that's reason for a custom `Persist`).